### PR TITLE
Make filter property animatable.

### PR DESCRIPTION
**Pull request by mantaroh**
This is a PR for https://bugzilla.mozilla.org/show_bug.cgi?id=1362897

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

There are tests for these changes, a test case will be landed in web-platform-tests in https://bugzilla.mozilla.org/show_bug.cgi?id=1362897

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17288)

### CODE REVIEW

**Comment by mantaroh** [comment]
This patche have been reviewed by @birtles, @heycam, @hiikezoe, @BorisChiou

**Comment by mantaroh** [comment]
Ah,  I removed unnecessary line before sending PR in order to build servo.(Because servo don't need the `use properties::longhands::filter::computed_value::Filter::*;`)
But this modification made unresolved the DropShadow when building the geckolib.

**Comment by nox** [comment]
> Ah, I removed unnecessary line before sending PR in order to build servo.(Because servo don't need the `use properties::longhands::filter::computed_value::Filter::*;`)
> But this modification made unresolved the DropShadow when building the geckolib.

You have to qualify the variants name by `Filter::` instead of glob-importing all of those enum's variants.

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -3518,6 +3512,64 @@ fn static_assert() {
...
+                            &filter.mFilterParameter).unwrap()));
+                },
+                NS_STYLE_FILTER_DROP_SHADOW => {
+                    let shadow_val;
+                    unsafe {
+                        let ref orig_shadow =
+                            (**filter.__bindgen_anon_1.mDropShadow.as_ref()).mArray[0];
+                        shadow_val = orig_shadow.to_shadow();
+                    };
+                    filters.push(DropShadow(shadow_val));
```
Comment (nox) [inline components/style/properties/gecko.mako.rs]: AFAICT you could just do:

```rust
filters.push(unsafe {
    (**filter.__bindgen_anon_1.mDropShadow.as_ref()).mArray[0].to_shadow()
});
```

Comment (mantaroh) [inline components/style/properties/gecko.mako.rs]: Thanks, @nox !

I addressed it, and changed to use `Filter::` in order to qualify the variant name.

**Comment by jdm** [comment]
```
  ▶ TIMEOUT [expected PASS] /css-animations-1_dev/html/css-filters-animation-blur.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected FAIL] /css-animations-1_dev/html/css-filters-animation-brightness.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected FAIL] /css-animations-1_dev/html/css-filters-animation-combined-001.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected FAIL] /css-animations-1_dev/html/css-filters-animation-contrast.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected PASS] /css-animations-1_dev/html/css-filters-animation-drop-shadow.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected FAIL] /css-animations-1_dev/html/css-filters-animation-grayscale.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected FAIL] /css-animations-1_dev/html/css-filters-animation-hue-rotate.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected FAIL] /css-animations-1_dev/html/css-filters-animation-invert.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected FAIL] /css-animations-1_dev/html/css-filters-animation-opacity.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected FAIL] /css-animations-1_dev/html/css-filters-animation-saturate.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ TIMEOUT [expected FAIL] /css-animations-1_dev/html/css-filters-animation-sepia.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1
```

**Comment by emilio** [comment]
Presumably those tests were broken before, and animations were just not running. The story to stop the compositor to take a screenshot in Servo when animations have finished is not great :(

**Comment by mantaroh** [comment]
I thought this cause is simply forgetting to remove meta file, but I faced test timeout even if remove meta file. I'm not sure why servo will be TIMEOUT..
If I understand correct, this reftest take a screenshot when load event firing, it looks like servo don't fire the load event..

I tried the `reftest-wait`, however the result is same timeout.

**Comment by mantaroh** [comment]
Umm,, If remove animation, reftest are success, and it is success when animation has finished. It looks like that unfinished animation caused reftest timeout. 

@emilio , @SimonSapin , @Manishearth 
The reftest with unfinished animation will cause timeout in Servo.
Do you know something about this phenomenon?

**Comment by mantaroh** [comment]
I want to separate this problem into another issue. I filed the issue #17335, in this PR, I'll skip those tests.

**Comment by birtles** [comment]
Looks like they're still timing out though.

**Comment by hiikezoe** [comment]
yeah, other tests in tests/wpt/css-tests/filters-1_dev.

File: tests/wpt/metadata-css/css-animations-1_dev/html/css-filters-animation-brightness.htm.ini [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [css-filters-animation-brightness.htm]
   type: reftest
-  expected: FAIL
+  disabled: The reftest of servo with non-expired animations failed(issue 17335).
```
Comment (jdm) [inline tests/wpt/metadata-css/css-animations-1_dev/html/css-filters-animation-brightness.htm.ini]: There is no need to disable these tests. We should use `expected: TIMEOUT` instead, and add a separate `bug: https://github.com/servo/servo/issues/17335` as well.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by mantaroh** [comment]
Thanks, @hiikezoe , @birtles , @jdm .
I overlooked that I need to change the ini file of filters-1_dev.
I updated init file of tests/wpt/css-tests/filters-1_dev and tests/wpt/css-tests/css-animations-1_dev by using `expected: TIMEOUT`.

**Comment by jdm** [comment]
```
  ▶ FAIL [expected TIMEOUT] /css-animations-1_dev/html/css-filters-animation-contrast.htm
  └   → /css-animations-1_dev/html/css-filters-animation-contrast.htm 0f0e0e7166718073fb1271121f89e4aecfbc750d
/css-animations-1_dev/html/reference/css-filters-animation-contrast-ref.htm 7e54b8d874be52ec34436f49bd8b3230e7f70e17
Testing 0f0e0e7166718073fb1271121f89e4aecfbc750d == 7e54b8d874be52ec34436f49bd8b3230e7f70e17

  ▶ PASS [expected TIMEOUT] /filters-1_dev/html/css-filters-animation-drop-shadow.htm
```
Oh good, these animation filter tests give us intermittent results, too!

**Comment by jdm** [comment]
I have filed issues so that they will be classified as intermittent failures. You can retry this PR without any changes.

**Comment by birtles** [comment]
Looks like a few more intermittents?

**Comment by jdm** [comment]
Updated.

**Comment by jdm** [comment]
Nevermind. It's clearly going to be a laborious process to keep the tests enabled; just disable them all instead (both the filter animation ones in /css-animations-1_dev/html/ and /filters-1_dev/html/).

**Comment by mantaroh** [comment]
Thanks @jdm, @birtles 
I updated those tests meta file with the disabled annotation.

**Comment by birtles** [comment]
I'm not sure if we use exactly the same format for these meta files as we do in Gecko, but if so I guess we just need a `__dir__.ini` that specifies `disabled: https://github.com/servo/servo/issues/17335`
(As per https://developer.mozilla.org/en-US/docs/Mozilla/QA/web-platform-tests#Per-directory_settings)

Also, let's drop the unrelated changes to MANIFEST.json

**Comment by mantaroh** [comment]
Thanks @birtles 
In those tests directory contain another tests, so I updated the disable annotation's comment, and removed the changes of metadata-css/MANIFEST.json.

**Pull request merged**